### ISSUE TITLE
Correctly produce unoptimized LLVM IR code based on LLVM version

### ISSUE THREAD

**Issue by compor** [compor/llvm-ir-cmake-utils#14]
There seems to have been a change in the way `-O0` works in conjunction with `--emit-llvm`.

If `-O0` is used with the latest LLVM versions, functions get annotated with `optnone` attribute, which prohibits further optimizations. The workaround seems to be to use `-O1` along with `-disable-llvm-optzns` and/or `-disable-llvm-passes`.

The related discussion can be found [here](https://reviews.llvm.org/D28047).

- [x] Determine minimum affected LLVM version
- [x] Determine LLVM commit number
- [x] Determine exact command line arguments required
- [ ] Resolve and test against various versions (having a test, it'd have been nice)

**Comment by compor** [comment]
From so far use it seems not to affect:

- LLVM 3.7
- LLVM 3.8

These can get unoptimized IR that **can be optimized** later with `--emit-llvm -O0`.

**Comment by compor** [comment]
The minimum affected version is:
- LLVM 5.0

according to this [thread][1] and added with this [review][2].

[1]: http://lists.llvm.org/pipermail/llvm-dev/2017-September/117534.html
[2]: https://reviews.llvm.org/D28404

**Comment by compor** [comment]
Since we simply add `-emit-llvm` to the already provided options, if we just want to get raw unoptimized IR, but with the ability to further optimize it (i.e. IR without the `optnone` attribute) we need to add the following options for `clang`:

```bash
-O1 -Xclang -disable-llvm-passes
```
or

```bash
-Xclang -disable-O0-optnone
```

Please note that `cmake` uses `-O1` if no optimization level is explicitly specified.

**Comment by compor** [comment]
Addressed in documentation in a7058f9fe5b662fce7076c857ff2c96ffee839b7 with version tag 2.6.0.
